sklearn/cross_validation.py

```diff
@@ -1820,8 +1821,13 @@ def train_test_split(*arrays, **options):
 
     Returns
     -------
-    splitting : list of arrays, length=2 * len(arrays)
-        List containing train-test split of input array.
+    splitting : length = 2 * len(arrays), 
+        If input is a list, the output will be a list.
+        If numpy array, the output will be a numpy array.
+        If scipy.sparse matrix, the output will be a 
+        scipy.sparse matrix. If input is a pandas dataframe, 
```

Comment:
Double checked and changed the description. I don't know if I did the `versionadded` thing correctly but I have added those lines. I will add this in the whatsnew of 0.16 API too.

Code to check the input output types for different input types.
```
from sklearn.cross_validation import train_test_split
import pandas as pd
import numpy as np
from scipy import sparse

matrix = np.array([[1, 2, 0],[0, 0, 3],[1, 0, 4]])
mat_tr, mat_test = train_test_split(matrix)

print type(matrix) == type(mat_tr) == type(mat_test)

sp = sparse.csr_matrix(matrix)
sp_tr, sp_test = train_test_split(sp)

print type(sp_tr) == type(sp) == type(sp_test)

lists = [[1, 2, 3], [4, 5, 6], [7, 8, 9]]
lists_tr, lists_test = train_test_split(lists)

print type(lists_tr) == type(lists) == type(lists_test)


data = {"a" : range(10), "b": range(10)}
df = pd.DataFrame(data)

df_tr, df_test = train_test_split(df)
print type(df) == type(df_tr) == type(df_test)
```